packages/twenty-front/src/modules/page-layout/widgets/graph/graphWidgetBarChart/utils/__tests__/calculateBarChartValueRange.test.ts

```diff
@@ -0,0 +1,287 @@
+
+      expect(result).toEqual({
+        min: 0,
+        max: 0,
+      });
+    });
+
+    it('should handle missing values (undefined)', () => {
+      const data: BarChartDataItem[] = [
+        { category: 'A', value1: 10 },
+        { category: 'B', value1: 20, value2: 30 },
+        { category: 'C', value2: 40 },
+      ];
+      const keys = ['value1', 'value2'];
+
+      const result = calculateBarChartValueRange(data, keys);
+
+      expect(result).toEqual({
+        min: 0,
+        max: 40,
+      });
+    });
+
+    it('should ignore NaN values', () => {
+      const data: BarChartDataItem[] = [
+        { category: 'A', value1: 10, value2: NaN },
+        { category: 'B', value1: 20, value2: 30 },
+      ];
+      const keys = ['value1', 'value2'];
+
+      const result = calculateBarChartValueRange(data, keys);
+
+      expect(result).toEqual({
+        min: 0,
+        max: 30,
+      });
+    });
+
+    it('should handle multiple series with different ranges', () => {
+      const data: BarChartDataItem[] = [
+        { category: 'A', sales: 1000, profit: -50, growth: 5.5 },
+        { category: 'B', sales: 2000, profit: 200, growth: -2.3 },
+        { category: 'C', sales: 1500, profit: 100, growth: 8.1 },
+      ];
+      const keys = ['sales', 'profit', 'growth'];
+
+      const result = calculateBarChartValueRange(data, keys);
+
+      expect(result).toEqual({
+        min: -50,
+        max: 2000,
+      });
+    });
+
+    it('should handle decimal values', () => {
+      const data: BarChartDataItem[] = [
+        { category: 'A', value1: 10.5, value2: -5.25 },
+        { category: 'B', value1: 15.75, value2: -10.5 },
+        { category: 'C', value1: 8.3, value2: -3.8 },
+      ];
+      const keys = ['value1', 'value2'];
+
+      const result = calculateBarChartValueRange(data, keys);
+
+      expect(result).toEqual({
+        min: -10.5,
+        max: 15.75,
+      });
+    });
+
+    it('should handle very large numbers', () => {
+      const data: BarChartDataItem[] = [
+        { category: 'A', value1: 1000000, value2: -500000 },
+        { category: 'B', value1: 2000000, value2: -1000000 },
+      ];
+      const keys = ['value1', 'value2'];
+
+      const result = calculateBarChartValueRange(data, keys);
+
+      expect(result).toEqual({
+        min: -1000000,
+        max: 2000000,
+      });
+    });
+  });
+
+  describe('real-world scenarios', () => {
+    it('should handle profit/loss financial data', () => {
+      const data: BarChartDataItem[] = [
+        { quarter: 'Q1', profit: -50000 },
+        { quarter: 'Q2', profit: -20000 },
+        { quarter: 'Q3', profit: 10000 },
+        { quarter: 'Q4', profit: 80000 },
+      ];
+      const keys = ['profit'];
+
+      const result = calculateBarChartValueRange(data, keys);
+
+      expect(result).toEqual({
+        min: -50000,
+        max: 80000,
+      });
+    });
+
+    it('should handle temperature data (can be negative)', () => {
```

Comment:
Not sure that those tests are relevant, let's test only the essential use cases while keeping a low footprint for maintenance.